### PR TITLE
Fix dev mode when you don't specify a configuration file

### LINKED ISSUE #1801: Running `--dev` with no `--config` doesn't enable dev mode
## Reproduction

1. Use [Starstuff](https://supergraph.demo.starstuff.dev/) as `starstuff.graphql`
2. Run
	  ```
	  ./router --supergraph starstuff.graphql --dev
	  ```

### Observed Behavior

```
INFO Apollo Router v1.0.0-rc.0 // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)
INFO GraphQL endpoint exposed at http://127.0.0.1:4000/ 🚀
```

### Expected Behavior

```
INFO Apollo Router v1.0.0-rc.0 // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)
WARN Development mode has been enabled and has not been started by `rover dev`. This mode of operation is only meant for development!
INFO enabled plugins: ["apollo.include_subgraph_errors", "apollo.csrf", "apollo.telemetry", "experimental.expose_query_plan"]
```

Currently, the implementation of `ConfigurationSource` only accepts `dev` on `ConfigurationSource::File`:

https://github.com/apollographql/router/blob/cf97c47e8bdabdea1409432b59021a247aea9ee9/apollo-router/src/router.rs#L241-L270

and even more specifically...

https://github.com/apollographql/router/blob/cf97c47e8bdabdea1409432b59021a247aea9ee9/apollo-router/src/router.rs#L267-L268

This is probably my mistake for leaning into this pattern I started with in #1748 (which I cargo culted from https://github.com/apollographql/router/pull/1475) which I handed off to @bnjjj.

### LINKED ISSUE #1802: `include_subgraph_errors` is always on by default
## Reproduction

1. Use [Starstuff](https://supergraph.demo.starstuff.dev/) as `starstuff.graphql`
2. Run without enabling dev mode and with no configuration:
	  ```
	  ./router --supergraph starstuff.graphql
	  ```

### Observed Behavior

```
INFO Apollo Router v1.0.0-rc.0 // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)
INFO enabled plugins: ["apollo.include_subgraph_errors", "apollo.csrf", "apollo.telemetry"]
INFO GraphQL endpoint exposed at http://127.0.0.1:4000/ 🚀
```

### Expected Behavior

```
INFO Apollo Router v1.0.0-rc.0 // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)
INFO enabled plugins: ["apollo.include_subgraph_errors", "apollo.csrf", "apollo.telemetry"]
INFO GraphQL endpoint exposed at http://127.0.0.1:4000/ 🚀
```

I would not expect to see `apollo.include_subgraph_errors` since I have provided no configuration and that is an option added in dev mode.

https://github.com/apollographql/router/blob/e9d187a2e555fa62077dfc248616857bec97d965/apollo-router/src/configuration/mod.rs#L279-L281

Maybe this is only because:

https://github.com/apollographql/router/blob/cf97c47e8bdabdea1409432b59021a247aea9ee9/apollo-router/src/router_factory.rs#L168-L170

?

Is this intentional?

### PR DESCRIPTION
**Pull request by bnjjj**
close #1801 

close #1802

### CODE REVIEW

**Review by garypen — APPROVED** [review]

**Review by o0Ignition0o — APPROVED** [review]

File: apollo-router/src/router.rs [hunk 1]
```diff
@@ -279,12 +279,20 @@ impl ConfigurationSource {
     /// Convert this config into a stream regardless of if is static or not. Allows for unified handling later.
     fn into_stream(self) -> impl Stream<Item = Event> {
         match self {
-            ConfigurationSource::Static(instance) => {
+            ConfigurationSource::Static(mut instance) => {
+                if instance.dev.unwrap_or_default() {
```
Comment (BrynCooke) [inline apollo-router/src/router.rs:283]: Possibly this logic should be here: https://github.com/apollographql/router/blob/main/apollo-router/src/configuration/mod.rs#L368

@o0Ignition0o WDTY? You were looking at this recently.

Comment (bnjjj) [reply apollo-router/src/router.rs:283]: Hmm still weird to have a mutation in the `validate` method. But I can add this validate call to my tests for dev mode. We will be more safe

Comment (BrynCooke) [reply apollo-router/src/router.rs:283]: Yeah, not the `validate` method.
But `ConfigurationSource` still feels like the wrong place. Checking

Comment (bnjjj) [reply apollo-router/src/router.rs:283]: I will open an issue regarding this validation problem. I tried several solutions but each of them has an issue. I need to dig more into it. I will merge this PR to solve the root issue, we also have `validate` called in the tests so we're safe for now but I agree we should find a better way to implement it

**Pull request merged**